RDKB-61783 : Telemetry NOT reporting when SecureElement is NOT accessible

adrojaankur opened the pull request:
Signed-off-by: Ankur Adroja <Ankur_Adroja@comcast.com>

adrojaankur committed:
Update curlinterface.c

adrojaankur committed:
Update xconfclient.c

shibu-kv reviewed (changes requested):
As per libcurl documentation,
CURLOPT_SSLENGINE | Loads/selects an SSL engine by name |
CURLOPT_SSLENGINE_DEFAULT | Sets the above engine as the default for SSL ops

Adding the second option only in cases of failure doesn't seem to be the right sequence.
This option should have been ideally selected in cases of success is what I interpret.
Could you please revisit the changes and experiment of the call stacks behavior.

shibu-kv committed:
RDKB-61783 : Fix overiding SSL options that is not accessible

Reason for change : Do not force override SSL engine selection.
  Use default options enabled from openssl

shibu-kv reviewed (approved):
Merging the changes reluctantly due to release timelines.
